lib/core/sdam/topology.js

```diff
@@ -142,7 +142,11 @@ class Topology extends EventEmitter {
       // Promise library
       promiseLibrary: options.promiseLibrary || Promise,
       credentials: options.credentials,
-      clusterTime: null
+      clusterTime: null,
+
+      // timer management
+      monitorTimers: [],
+      iterationTimers: []
```

Comment:
These would be better as `Set`s. It would let you do

```js
for (const timer of monitorTimers) {
  clearTimeout(timer);
}
monitorTimers.clear()
```

and

```js
monitorTimers.delete(timer)
```